cmd/karavictl/cmd/storage_list.go

```diff
@@ -63,12 +80,34 @@ var listCmd = &cobra.Command{
 		if err := yaml.Unmarshal(scrubbed, &m); err != nil {
 			reportErrorAndExit(JSONOutput, cmd.ErrOrStderr(), err)
 		}
-		if err := JSONOutput(cmd.OutOrStdout(), &m); err != nil {
+
+		s := filterStorage(storageType, m)
+		if err := JSONOutput(cmd.OutOrStdout(), &s); err != nil {
 			reportErrorAndExit(JSONOutput, cmd.ErrOrStderr(), err)
 		}
 	},
 }
 
+func filterStorage(storageType string, allStorage map[string]interface{}) interface{} {
+	if storageType == "" {
+		return allStorage
+	}
+
+	output := make(map[string]interface{})
+	if storage, ok := allStorage["storage"].(map[string]interface{}); ok {
```

Comment:
Optional: you could reduce most of the indentation of this code by negating the `if`s and utilizing `continue`.

Reply:
Refactored.